coalib/settings/ConfigurationGathering.py

```diff
@@ -109,11 +109,18 @@ def warn_nonexistent_targets(targets, sections, log_printer):
     :param sections:    The sections to search. (Dict.)
     :param log_printer: The log printer to warn to.
     """
+
+    sections_substrings = {
+        substring
+        for sections_name in sections
+        for substring in sections_name.split('.')
+    }
     for target in targets:
-        if target not in sections:
+        if target not in sections_substrings | set(sections):
```

Comment:
I'd rather have the logic of the get_substring_targets somehow in here than below adding artificial targets, this should change the interpretation of the targets and not define new ones IMO